javaparser-core/src/main/javacc/java_1_8.jj

```diff
@@ -256,13 +261,39 @@ SKIP :
 TOKEN_MGR_DECLS :
 {
     private List<Token> tokens = new ArrayList<Token>();
+    private CommentsCollection commentsCollection = new CommentsCollection();
 
     List<Token> getTokens() {
         return tokens;
     }
+    
+    CommentsCollection getCommentsCollection() {
+        return commentsCollection;
+    }
 
     private void CommonTokenAction(Token token) {
         tokens.add(token);
+
+        while (token.specialToken != null) {
```

Comment:
A few lines seems long (at least in the split view I am using for this review). For example line 282 or 292.

There a few places where a space is missing around operators (e.g., lines 282 and 284)